l3build-upload.lua

```diff
@@ -245,8 +245,12 @@ end
 function shell(s)
   local h = assert(popen(s, 'r'))
   local t = assert(h:read('*a'))
-  h:close()
-  return t
+  local success = h:close()
+  if (success) then
+   return t
+  else
+   error("\nError from shell command:\n" .. s .. "\n" .. t .. "\n")
+  end
```

Comment:
```suggestion
  h:close() or error("\nError from shell command:\n" .. s .. "\n" .. t .. "\n")
  return t
```

Reply:
@hmenke  more idiomatic Lua I guess, happy to change to that

Reply:
@hmenke 

```
error loading module ./l3build-upload.lua from file ./l3build-upload.lua:
        ./l3build-upload.lua:248: unexpected symbol near 'or'

```

it doesn't seem to like inlining h:close (which has three return values) here?

Reply:
Nevermind then, it was just a thought.